akka-http-testkit/src/test/java/akka/http/javadsl/testkit/JUnitRouteTestTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.javadsl.testkit;
+
+import java.util.function.Function;
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.Test;
+import akka.http.javadsl.testkit.*;
+
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.model.headers.*;
+import akka.http.javadsl.server.Rejections;
+
+public class JUnitRouteTestTest extends JUnitRouteTest {
+
+  @Test
+  public void testTheMostSimpleAndDirectRouteTest() {
+    TestRoute route =
+      testRoute(
+        complete(HttpResponse.create())
+      );
+
+    route.run(HttpRequest.GET("/"))
+      .assertStatusCode(StatusCodes.OK);
+  }
+
+  @Test
+  public void testUsingADirectiveAndSomeChecks() {
+    RawHeader pinkHeader = RawHeader.create("Fancy", "pink");
+
+    TestRoute route =
+      testRoute(
+        respondWithHeader(pinkHeader, () ->
+          complete("abc")
+        )
+      );
+
+    route.run(HttpRequest.GET("/").addHeader(pinkHeader))
+      .assertStatusCode(StatusCodes.OK)
+      .assertContentType("text/plain; charset=UTF-8")
+      .assertEntity("abc")
+      .assertHeaderExists("Fancy", "pink");
+  }
+
+  @Test
+  public void testProperRejectionCollection() {
+    TestRoute route =
+      testRoute(
+        get(() ->
+          complete("naah")
+        ).orElse(
+          put(() ->
+            complete("naah")
+          )
+        )
+      );
+
+    route.run(HttpRequest.POST("/abc").withEntity("content"))
+      /* XXX: Expected?
+      .assertRejections(
+        Rejections.method(HttpMethods.GET),
+        Rejections.method(HttpMethods.PUT)
+      )
```

Comment:
That does seem a bit verbose - for test cases where the request isn't rejected it doesn't matter if you're running sealed or unsealed, and it'd be kind of a shame to make that so prominent there.

Ideally I think `run` would run unsealed (consistent with scaladsl) and we'd add a `runSealed` to run sealed. I'm not sure how much of our users already use this DSL though, so perhaps that ship has sailed - might be OK to just add `runUnsealed` next to `run` then though?